src/client/src/Stream.c

```diff
@@ -2120,11 +2127,12 @@ STATUS resetCurrentViewItemStreamStart(PKinesisVideoStream pKinesisVideoStream)
         pKinesisVideoStream->curViewItem.offset = overallSize;
     }
 
-    // Set the new length
+    // Set the new length in the view
     pViewItem->length = overallSize;
 
-    // Set the handle that will need to be freed on exit - now we should free the old one
-    allocationHandle = oldAllocationHandle;
+    // No need to set the size of the actual allocation - just unmap
```

Comment:
I don't think we need to unmap here since there's another unmap at L2158.

Reply:
True, however, right after we unmap we set the pFrame to NULL which will make the cleanup unmapping a no-op. I generally prefer doing the cleanup if there is more stuff in the body of the function following it before the cleanup. In this case, it's nice to do that as we hold the lock. In fact, we should re-acquire the lock to do the unmapping. Will change